### ISSUE TITLE
🐞[Bug] metadata_scan crashes on Linux Docker

### ISSUE THREAD

**Issue by David-Maisonave** [stg-annon/stashapi#16]
When running Stash on Linux in Docker, metadata_scan crashes, because **scan** is set to **None**.
The following line fails:
`
scan_metadata_input.update(self.get_configuration_defaults("scan { ...ScanMetadataOptions }").get("scan",{}))
`

The following is a fix that worked in both Windows in Linux Docker:
``` python
    def metadata_scan(self, paths:list=[], flags={}):
        query = "mutation MetadataScan($input:ScanMetadataInput!) { metadataScan(input: $input) }"
        scan_metadata_input = {"paths": paths}
        if flags:
            scan_metadata_input.update(flags)
        else:
            scanData = self.get_configuration_defaults("scan { ...ScanMetadataOptions }")
            if scanData['scan'] != None:
                scan_metadata_input.update(scanData.get("scan",{}))
        result = self.call_GQL(query, {"input": scan_metadata_input})
        return result["metadataScan"]

```

**Comment by stg-annon** [comment]
hmm shouldn't really be any different getting the default config

I would expect any issues to be more with escaping the paths and passing them to stash

try using this
```py
def metadata_scan(self, paths:list=[], flags={}):
	query = "mutation MetadataScan($input:ScanMetadataInput!) { metadataScan(input: $input) }"
	scan_metadata_input = {"paths": paths}
	if flags:
		scan_metadata_input.update(flags)
	elif scan_config := scan_metadata_input.update(self.get_configuration_defaults("scan { ...ScanMetadataOptions }").get("scan")):
		scan_metadata_input.update(scan_config)
	result = self.call_GQL(query, {"input": scan_metadata_input})
	return result["metadataScan"]
```

**Comment by David-Maisonave** [comment]
I'll try it as soon as I have my Linux Docker available again.

**Comment by David-Maisonave** [comment]
That change still produces the same error.
```
  File "/root/.stash/plugins/community/renamefile/StashPluginHelper.py", line 523, in metadata_scan
    elif scan_config := scan_metadata_input.update(self.get_configuration_defaults("scan { ...ScanMetadataOptions }").get("scan")):
                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: 'NoneType' object is not iterable
```

**Comment by stg-annon** [comment]
just realized I forgot to delete part of that try
```py
def metadata_scan(self, paths:list=[], flags={}):
	query = "mutation MetadataScan($input:ScanMetadataInput!) { metadataScan(input: $input) }"
	scan_metadata_input = {"paths": paths}
	if flags:
		scan_metadata_input.update(flags)
	elif scan_config := self.get_configuration_defaults("scan { ...ScanMetadataOptions }").get("scan"):
		scan_metadata_input.update(scan_config)
	result = self.call_GQL(query, {"input": scan_metadata_input})
	return result["metadataScan"]
```

**Comment by David-Maisonave** [comment]
That version works on Docker Linux.

**Comment by stg-annon** [comment]
resolved with 0e98344b0a285bcc62e18790366dc5c1a0aab8af
